home/rvagg/2021-02-exploration-report-small-filecoin-deals.md

```diff
@@ -0,0 +1,134 @@
+
+Filecoin currently only stores ~5 Pb of deal data and growth rate is not large.
+
+## Installing and Running Lotus
+
+Followed instructions at https://docs.filecoin.io/get-started/lotus/installation/ for an AWS instance
+
+ * Large AMD EPYC instance, Ubuntu 20.04
+ * Installed dependencies (used Go Snap rather than Go tarball)
+ * Cloned and build Lotus with CPU SHA support
+ * Installed systemd service to run as user `lotus` and put logs into that home directory
+ * Used minimal stateroots snapshot @ https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car
+ * Started manually with `lotus daemon --import-snapshot`
+ * Stopped, then started via systemd, let it sync, watching the log output with `tail -f`
+ * Can interact with the daemon via the CLI with `lotus client` and `lotus wallet`, etc. as the `lotus` user (appears to be user-restricted IPC)
+ * Made a wallet and transferred some FIL in it for experimenting
+
+Interesting commands:
+
+ * `lotus sync wait` will run until the chain is synced, zero exit status on success
+ * `lotus wallet` - `lotus wallet new`, `lotus wallet new`, `lotus wallet balance`, `lotus wallet export` (export privkey for backup)
+ * `lotus client`
+   * `lotus client import <path>` to import file(s) into the client (somewhere?), presumably as UnixFS (?), gets a CID
+   * `lotus client local` shows you what you have imported and their CIDs
+   * `lotus client balances` shows you funds that you have active in the "market" (where you have to have funds in order to make deals etc. i.e. out of your wallet)
+   * `lotus client find <CID>` lets you query the network for a CID
+
+Deals:
+
+   * `lotus client query-ask <minerID>` ask a miner for their deal terms - returns FIL asking price per GiB per epoch (30 seconds) for both regular and verified deals, along with minimum and maximum deal sizes.
+   * `lotus client list-asks` will do a scan of the network for miners with power and do a `query-ask` on each of them, reporting the output to stdout, one miner per line (~450)
+   * `lotus client deal` without additional arguments takes you through a prompted deal-making process that asks for your data CID, a miner ID, whether it's a verified deal; it queries the miner, asks you to confirm the asking price and starts the deal process if you say `yes`.
+   * `lotus client list-deals` to show you deals in progress and their status (Note: the status identifiers are not self-explainatory, the best source of documentation seems to be the markets code: https://github.com/filecoin-project/go-fil-markets/blob/master/storagemarket/dealstatus.go)
+   * `lotus client deal-stats` seems to be the best way to view the outcome of your deals once they disappear off `list-deals`
```

Comment:
> once they disappear 

There's the oddly named `--show-failed` flag that "brings them back"
There's also `-v` to that command